hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -324,6 +324,13 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 	}
 
 	if err := r.ntoReconcile(ctx, nodePool, configGenerator, controlPlaneNamespace); err != nil {
+		SetStatusCondition(&nodePool.Status.Conditions, hyperv1.NodePoolCondition{
```

Comment:
ntoReconcile is already setting NodePoolValidTuningConfigConditionType for some error cases, which should not set this condition too.
Please this condition from within the function only for the cases where is relevant

Reply:
Done.
I used the `hyperv1.NodePoolValidMachineConfigConditionType` as the condition type.
If you think there's a more suitable type, please let me know.